src/App.vue

```diff
@@ -89,8 +89,11 @@ export default {
 
     this.$store.dispatch('app/FETCH_VERIFIED_USERS')
 
-    EventBus.$on('socket:codex-coin:transferred', () => {
+    // @TODO: sometimes this doesn't update the balance with the latest
+    //  balance... almost as if web3 or infura is caching the balanceOf call
+    EventBus.$on('socket:codex-coin:transferred', (codxCost) => {
```

Comment:
We should fetch the balance via API now instead of from the contract. Then we just have to keep it up to date when socket events come through (or just always pull from the API again).

Reply:
That is what's happening here. The API-managed balance is show in the sidebar, and we dynamically adjust as CODX is spent (similar to how you did it with gas before.) This comment is referring to the `FETCH_TOKEN_BALANCE` call, which I left in place because it's still affecting certain pages (e.g the faucet page) and doesn't really hurt to just leave. But I suppose it could be removed.... I was thinking for savvy users we'd still want to read from the contract. Let's discuss on our call today.